Makefile

```diff
@@ -62,6 +62,10 @@ ifdef CUDA_FLAGS
 CMAKE_FLAGS += -DCMAKE_CUDA_FLAGS="$(CUDA_FLAGS)"
 endif
 
+ifdef USE_CLANG
```

Comment:
nit: Can you check if doing something like setting CC, CXX works ? If it does we can remove this.

Reply:
Sure. For the setup scripts we can keep this? 
The issue is that some setup scripts override the compiler in their own right (e.g. Centos9 sourcing gcc12). I suppose we could check for CC and CXX and not override anything else if set.